app_dart/lib/src/service/scheduler.dart

```diff
@@ -143,12 +146,70 @@ class Scheduler {
       return;
     }
 
+    final slug = pr.base!.repo!.slug();
     final String fullRepo = pr.base!.repo!.fullName;
     final String? branch = pr.base!.ref;
     final String sha = pr.mergeCommitSha!;
 
     final String id = '$fullRepo/$branch/$sha';
     final Key<String> key = datastore.db.emptyKey.append<String>(Commit, id: id);
+
+    // In non-merge queue cases, the `mergedAt` time for closed pull request was
+    // enough to correctly identify the order of linear history.
+    //
+    // However, in a merge queue, commits can get held up by slower running
+    // tests on commits ahead in the queue. When this occurs, all commits waiting
+    // are merged at the exact same time. This will confuse the dashboard and
+    // backfill scheduler.
+    //
+    // Solution in both cases: use graphql to query the time `committedDate`.
```

Comment:
If multiple commits are held up in the queue then landed simultaneously with identical timestamps, could we have multiple PRs enter the queue simultaneously with the same `committedDate`? Say your queue limit is 5, and GitHub lands a batch of 3 commit on master. There are now 3 free slots in the queue. GitHub will now go ahead an add 3 PRs to the queue simultaneously, creating commits for them at the same time, with the same `commttedDate` (according to the interwebs git commit date precision is 1 second, which is way too coarse-grained).